cupyx/scipy/sparse/data.py

```diff
@@ -73,6 +73,37 @@ def count_nonzero(self):
         """
         return cupy.count_nonzero(self.data)
 
+    def mean(self, axis=None, dtype=None, out=None):
+        """Compute the arithmetic mean along the specified axis.
+
+        Args:
+            axis (int or ``None``): Axis along which the sum is comuted.
+                If it is ``None``, it computes the average of all the elements.
+                Select from ``{None, 0, 1, -2, -1}``.
+
+        Returns:
+            cupy.ndarray: Summed array.
+
+        .. seealso::
+           :meth:`scipy.sparse.spmatrix.mean`
+
+        """
+        validateaxis(axis)
+        nRow, nCol = self.shape
+        data = self.data.copy()
```

Comment:
```python
n = ...
data = cupy.true_divide(self.data, n, dtype=dtype)
return self._with_data(data).sum(axis, dtype, out)
```
looks good.  (It seems the `_with_data` method can receive `data` of different `dtype`.)

Reply:
I just followed what `power` did. Do I need to change my code?

Reply:
In `power` copies the input and overwrites the new array.  To follow it, `data /= n` after the copy.  Python 2 support has been dropped.

Reply:
`data` is copied and `self._with_data(data)` is used to sum, `_with_data` returns a new matrix right? Do you mean I should replace `true_divide` with `/=`?

Reply:
@toslunar What is the implication of no Python 2 support?

Reply:
> `data` is copied and `self._with_data(data)` is used to sum, `_with_data` returns a new matrix right?

Right.  There's only a performance issue that the original code allocates memory twice.

> Do you mean I should replace `true_divide` with `/=`?

Yes, (but there's another way).  It reduce the number of memory allocations.  `data /= n` (or `divide(data, n, out=data)`) is different from `data = data / n` (or `data = divide(data, n)`).
- If you choose `data /= n`, cast `data` to `dtype` first.
- What I said is `data = cupy.true_divide(self.data, n, dtype=dtype)` seems OK because `self.data` is kept.

> What is the implication of no Python 2 support?

`divide` or `/` can be used in place of `true_divide`.

Reply:
@toslunar It seems that the tests are passing after the latest commit because `sum` outputs dtype correctly